### PR TITLE
fix: `rpcIdentifierUtility` client side grouping before emitting CustomRPC event

### PR DESCRIPTION
**Pull request by gambinish**
## **Description**

Rather than arbitrarily sending `sha256` hashes over the wire, we should instead handle the lookup/grouping of custom rpcs client-side, and then send over the meaningful resource identifiers of commonly used.

Benefits are:
1. No post processing necessary
2. Bypasses the need to hash at all, since `rpcIdentifierUtility` should only return the rpc host when there's a match (not for private eth nodes, for instance)

Note:

1. `useSafeChains` hook is inspired by to be merged implementation [here](https://github.com/MetaMask/metamask-extension/blob/brian/salim-build-test/ui/pages/settings/networks-tab/networks-form/use-safe-chains.ts)

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26266?quickstart=1)

## **Related issues**

Fixes: MMASSETS-298

## **Manual testing steps**

1. Go to settings and add a Custom network from the custom network form
2. Event will be emitted with `sensitiveProperties` once network gets added

## **Screenshots/Recordings**

N/A UX behavior should be the same

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by legobeat — DISMISSED** [review]
LGTM!

File: ui/pages/settings/networks-tab/networks-form/use-safe-chains.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  safeChains.forEach((chain) => {
+    chain.rpc.forEach((rpc) => {
+      const sanitizedRpc = new URL(rpc).host;
+      knownRpcUrls.add(sanitizedRpc);
+    });
+  });
+
+  const sanitizedRpc = new URL(rpcUrl).host;
+  return knownRpcUrls.has(sanitizedRpc) ? sanitizedRpc : 'Unknown rpcUrl';
```
Comment (bergeron) [inline ui/pages/settings/networks-tab/networks-form/use-safe-chains.ts:58]: If we just care about finding a match, it seems like we can save memory by not storing all the hosts in a `Set`.   Plus short circuit as soon as we find one.  Something like:

```ts
const host = new URL(rpcUrl).host;

for (const chain of safeChains) {
  for (const rpc of chain.rpc) {
    if (host === new URL(rpc).host) {
      return host;
    }
  }
}

return 'Unknown rpcUrl';
```

Comment (gambinish) [reply ui/pages/settings/networks-tab/networks-form/use-safe-chains.ts]: https://github.com/MetaMask/metamask-extension/pull/26266/commits/3a12e1c8ffe6b9d77bbad8796cd4b5b06c3486fb

**Comment by gambinish** [comment]
Re-requesting after merging `develop` branch. Needs to be re-approved 🙏

**Review by legobeat — APPROVED** [review]

**Review by salimtb — APPROVED** [review]

**Pull request merged**